client/components/SearchBar.jsx

```diff
@@ -31,16 +31,22 @@ const SearchForm = styled.input`
   height: 31px;
   width: 100%;
   padding: 3px 25px 3px 29px;
-  cursor: default;
+  cursor: text;
   font-weight: 400l
 `;
 
-const SearchBar = () => {
-  const [searchTerm, setSearchTerm] = useState('');
-  const handleChange = event => {
-    console.log(event.target.value)
-    setSearchTerm(event.target.value);
-  };
+const SearchBar = ({ handleSearch }) => {
+  const [query, setQuery] = useState('');
+  const isMounted = useRef(false);
+
+  useEffect(() => {
```

Comment:
I'm a bit lost what this code is trying to do. Can you walk me through what lines 42 - 49 are doing? What is this code trying to accomplish?

Reply:
My goal is to call the `handleSearch` function within the `useEffect` hook after 500ms when user types into search bar. I am using `isMounted` flag to hold the values true and false to prevent useEffect from calling `handleSearch` immediately after mounting.

On page load, SearchBar is mounted and `isMounted` is set to true. As the component's state changes with each keypress, `useEffect` calls `handleSearch` after 500ms, and clears the timeout.

Without the `isMounted` flag, `handleSearch` would be called on mount with the default query state (`''`), and I wanted to prevent this.

Reply:
Makes sense!